--- a/buildbot/changes/gitpoller.py
+++ b/buildbot/changes/gitpoller.py
@@ -35,7 +35,7 @@
                  workdir=None, pollInterval=10*60, 
                  gitbin='git', usetimestamps=True,
                  category=None, project=None,
-                 pollinterval=-2):
+                 pollinterval=-2, encoding='utf-8'):
         # for backward compatibility; the parameter used to be spelled with 'i'
         if pollinterval != -2:
             pollInterval = pollinterval
@@ -44,6 +44,7 @@
         self.repourl = repourl
         self.branch = branch
         self.pollInterval = pollInterval
+        self.encoding = encoding
         self.lastChange = time.time()
         self.lastPoll = time.time()
         self.gitbin = gitbin
@@ -104,7 +105,7 @@
         return d
 
     def _get_commit_comments_from_output(self,git_output):
-        stripped_output = git_output.strip()
+        stripped_output = git_output.strip().decode(self.encoding)
         if len(stripped_output) == 0:
             raise EnvironmentError('could not get commit comment for rev')
         self.commitInfo['comments'] = stripped_output
@@ -148,7 +149,7 @@
         return d
 
     def _get_commit_name_from_output(self, git_output):
-        stripped_output = git_output.strip()
+        stripped_output = git_output.strip().decode(self.encoding)
         if len(stripped_output) == 0:
             raise EnvironmentError('could not get commit name for rev')
         self.commitInfo['name'] = stripped_output
--- a/docs/cfg-changesources.texinfo
+++ b/docs/cfg-changesources.texinfo
@@ -1237,6 +1237,12 @@
 
 @item usetimestamps
 parse each revision's commit timestamp (default is @code{True}), or ignore it in favor of the current time (so recently processed commits appear together in the waterfall page)
+
+@item encoding
+Set encoding will be used to parse author's name and commit message. Default
+encoding is @code{'utf-8'}. This will not be applied to file names since git
+will translate non-ascii file names to unreadable escape sequences.
+
 @end table
 
 @heading Example
